lib/yamllint/linter.rb

```diff
@@ -151,28 +151,30 @@ def parse(psych_parse_data)
       def parse_recurse(psych_parse_data, is_sequence = false)
         is_key = false
         psych_parse_data.children.each do |n|
-          case n.class.to_s
-          when 'Psych::Nodes::Scalar'
+          case n
```

Comment:
Shouldn't this be `n.class`?

Reply:
It's fine, though it's a mildly surprising idiom. It's designed to work this way: `case...when` in Ruby compares with `===`, and [`Module#===`](https://ruby-doc.org/core-2.5.3/Module.html#method-i-3D-3D-3D) is defined to check whether the object is an instance of the class.